api/http.py

```
@@ -0,0 +1,61 @@
+    def request(self, method, url, *args, **kwargs):
+        retryable = kwargs.pop("retryable", method.upper() in SAFE_RETRY_METHODS)
+        last_exc = None
+
+        for attempt in range(self._retries + 1):
+            try:
+                return super().request(method, url, *args, **kwargs)
+            except BrowserRequestException as exc:
+                last_exc = exc
+                if not retryable or attempt >= self._retries:
+                    raise
+                sleep_time = self._retry_backoff * (2 ** attempt) + random.uniform(0, self._retry_backoff)
```

Ruff (0.15.6)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)